File: DietAI-backend/src/main/java/com/backend/DietAIbackend/dto/RecipeInDiet.java
package com.backend.DietAIbackend.dto;

import com.backend.DietAIbackend.model.DayWeek;
import com.backend.DietAIbackend.model.MealTime;
import com.backend.DietAIbackend.model.Recipe;

import java.time.DayOfWeek;

public record RecipeInDiet(Recipe recipe, DayWeek dayWeek, MealTime mealTime) {
}

File: DietAI-frontend/src/app/models/RecipeInDiet.ts
import { Recipe } from './Recipe';
export class RecipeInDiet {
    recipe: Recipe = new Recipe();
    dayOfWeek: string = '';
    mealTime: string = '';
}

export class RecipeInDietResponse {
    recipe: Recipe = new Recipe();
    day: string = '';
    mealTime: string = '';
}